File: gonggzy_data/ggzy.js
const crypto = require("crypto-js")
var MD5 = require('md5.js')


let t={ts: 1684844032188, type: '12', IS_IMPORT: 1, pageSize: 3}
function s(n) {
                        return new MD5(!0).update(n).digest("hex")
                    }


function u(t, e) {
            return t.toString().toUpperCase() > e.toString().toUpperCase() ? 1 : t.toString().toUpperCase() == e.toString().toUpperCase() ? 0 : -1
        }
//console.log(s("3637CB36B2E54A72A7002978D0506CDFIS_IMPORT1pageSize3ts1684844032188type12"))
 function l(t) {
            for (var e = Object.keys(t).sort(u), n = "", a = 0; a < e.length; a++)
                if (void 0 !== t[e[a]])
                    if (t[e[a]] && t[e[a]]instanceof Object || t[e[a]]instanceof Array) {
                        var i = JSON.stringify(t[e[a]]);
                        n += e[a] + i
                    } else
                        n += e[a] + t[e[a]];
            return n
        }
function d(t) {
            for (var e in t)
                "" !== t[e] && void 0 !== t[e] || delete t[e];
            var n = '3637CB36B2E54A72A7002978D0506CDF' + l(t);
            return s(n).toLocaleLowerCase()
        }


function b(t) {
            var e = crypto.enc.Utf8.parse('BE45D593014E4A4EB4449737660876CE')
              , n = crypto.enc.Utf8.parse('A8909931867B0425')
              , a = crypto.AES.decrypt(t, e, {
                iv: n,
                mode: crypto.mode.CBC,
                padding: crypto.pad.Pkcs7
            });
            return a.toString(crypto.enc.Utf8)
        }

console.log(d({
    "ts": 1684845416020,
    "pageNo": 1,
    "pageSize": 20,
    "total": 0,
    "AREACODE": "",
    "M_PROJECT_TYPE": "",
    "KIND": "GCJS",
    "GGTYPE": "1",
    "PROTYPE": "",
    "timeType": "6",
    "BeginTime": "2022-11-23 00:00:00",
    "EndTime": "2023-05-23 23:59:59",
    "createTime": []
}))

File: gonggzy_data/ggzy.py
import time

import execjs

import requests

ctx = execjs.compile(open("ggzy.js", "r").read())

date = round(time.time() * 1000)

json_data = {
    'pageNo': 1,
    'pageSize': 20,
    'total': 0,
    'AREACODE': '',
    'M_PROJECT_TYPE': '',
    'KIND': 'GCJS',
    'GGTYPE': '1',
    'PROTYPE': '',
    'timeType': '6',
    'BeginTime': '2022-11-23 00:00:00',
    'EndTime': '2023-05-23 23:59:59',
    'createTime': [],
    'ts': date,
}

headers = {
    'Accept': 'application/json, text/plain, */*',
    'Accept-Language': 'en-US,en;q=0.9,zh-CN;q=0.8,zh;q=0.7',
    'Cache-Control': 'no-cache',
    'Connection': 'keep-alive',
    'Content-Type': 'application/json;charset=UTF-8',
    'Origin': 'https://ggzyfw.fujian.gov.cn',
    'Pragma': 'no-cache',
    'Referer': 'https://ggzyfw.fujian.gov.cn/business/list/',
    'Sec-Fetch-Dest': 'empty',
    'Sec-Fetch-Mode': 'cors',
    'Sec-Fetch-Site': 'same-origin',
    'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/113.0.0.0 Safari/537.36',
    'portal-sign': ctx.call("d",json_data),
    'sec-ch-ua': '"Google Chrome";v="113", "Chromium";v="113", "Not-A.Brand";v="24"',
    'sec-ch-ua-mobile': '?0',
    'sec-ch-ua-platform': '"Windows"',
}



response = requests.post('https://ggzyfw.fujian.gov.cn/FwPortalApi/Trade/TradeInfo', headers=headers, json=json_data)
print(headers['portal-sign'])

print(response.text)


print(ctx.call("b",response.json()["Data"]))


# Note: json_data will not be serialized by requests
# exactly as it was in the original request.
#data = '{"pageNo":1,"pageSize":20,"total":0,"AREACODE":"","M_PROJECT_TYPE":"","KIND":"GCJS","GGTYPE":"1","PROTYPE":"","timeType":"6","BeginTime":"2022-11-21 00:00:00","EndTime":"2023-05-21 23:59:59","createTime":[],"ts":1684674757813}'
#response = requests.post('https://ggzyfw.fj.gov.cn/FwPortalApi/Trade/TradeInfo', headers=headers, data=data)